### DISCUSSION TITLE
Altitude Offset for Flowdeck V2 with CF Bolt

### DISCUSSION THREAD

**Discussion by johnnwallace** [bitcraze/discussions#157] [Q&A] [answered]
I'm using a Flowdeck V2 on a CF Bolt platform using the F450 drone frame with legs. As such, the z ranger on the deck is much higher off the ground than it would typically be when using a CF 2.1. I was wondering if there was a way to add an offset parameter so that even if the deck is ~0.1 m off the ground, the drone's altitude would be interpreted as ~0 m. If that functionality is not built into the firmware by default, what would the best location be in the code to add that?

Thanks!

**Comment by knmcguire** [comment, marked as answer]
That is being handled from the [zranger2 driver code](https://github.com/bitcraze/crazyflie-firmware/blob/46f497a1ef058750f59277c88578f9f2c13ea670/src/deck/drivers/src/zranger2.c#L141-L143). Before it is sent to the Kalman filter, you can add the 0.1 meters to the distance like this? I've added the DISTANCE_OFFSET here so that you can make that a define on top of the file. 

      float distance = (float)range_last * 0.001f + DISTANCE_OFFSET; 
      float stdDev = expStdA * (1.0f  + expf( expCoeff * (distance - expPointA)));
      rangeEnqueueDownRangeInEstimator(distance, stdDev, xTaskGetTickCount());
